client/src/app/routes/dataset/code-editor/code-editor.component.html

```diff
@@ -0,0 +1,162 @@
+                    fxFlex
+                >
+                <tag-picker
+                    type="expression"
+                    [selectedTagIDs]="selectedTagIDs"
+                    [showCurrentTagsSection]="true"
+                    (onTagSelectionChanged)="onTagSelectionChanged($event)"
+                    [editable]="editable"
+                >
+                </tag-picker>
+            </div>
+        </div>
+        <div class="viz-container">
+            <span fxLayout="row" fxLayoutAlign="center center">
+                <h2>Visualization{{displayExpressionTitle ? ": " + displayExpressionTitle : ""}}</h2>
+                <icon-button 
+                    *ngIf="isCodeChanged"
+                    icon='assets/button-icons/disabled-orange.svg'
+                    [hasBackground]="false"
+                    #tooltip="matTooltip"
+                    matTooltip="Expression has changed since last run"
+                ></icon-button>
+            </span>
+            <div class="right-btn-group">
+                <icon-button
+                    icon='assets/button-icons/close-cross-gray.svg'
+                    (click)="onClose()"
+                    [hasBackground]="false"
+                ></icon-button>
+            </div>
+        </div>
+    </div>
+    <div fxLayout="row" class="editor-body">
+        <div class="text-editor">
+            <expression-text-editor
+                *ngIf="expression"
+                [expression]="expression"
+                [allowEdit]="editable"
+                [isImmediatelyAppliable]="false"
+                [applyNow]="false"
+                (onTextChange)="onExpressionTextChanged($event)"
+                (onTextSelect)="onTextSelect($event)"
+                (runExpression)="runExpression()"
+                (runHighlightedExpression)="runHighlightedExpression()"
+                (saveExpression)="onSave()"
+                [range]="isSubsetExpression ? executedTextSelection.range : null"
+                [showHelp]="useAutocomplete && !isSharedByOtherUser"
+                [isLua]="isLua"
+            >
+            </expression-text-editor>
+        </div>
+
+        <div class="viz-panels">
+            <div class="preview-container {{isPMCDataGridSolo ? 'hidden' : ''}}">
+                <template #preview></template>
+            </div>
+            <pmc-data-grid
+                class="{{isPMCDataGridSolo ? 'solo' : ''}}"
+                [header]="pmcGridExpressionTitle"
+                [evaluatedExpression]="evaluatedExpression"
+                [columnCount]="15"
+                (onToggleSolo)="onTogglePMCDataGridSolo($event)"
+            ></pmc-data-grid>
+        </div>
+    </div>
+    <div class="editor-footer">
+        <div class="text-container">
+            <mat-select [(ngModel)]="isLua">
+                <mat-option [value]="true">Lua</mat-option>
+                <mat-option [value]="false">Pixlang</mat-option>
```

Comment:
😆 Pixlang, why not!

Reply:
Happy to change it, but I was using `Pixlise Expression Language` before and that truncated the text weird after it was selected so needed something shorter 😅